OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -18,30 +20,36 @@ namespace OpenRA.Mods.Common.Activities
 	public class FlyAttack : Activity
 	{
 		readonly Target target;
+		readonly AttackPlane attackPlane;
+		readonly IEnumerable<AmmoPool> ammoPools;
 		Activity inner;
 		int ticksUntilTurn = 50;
 
-		public FlyAttack(Target target) { this.target = target; }
+		public FlyAttack(Actor self, Target target)
+		{
+			this.target = target;
+			attackPlane = self.TraitOrDefault<AttackPlane>();
+			ammoPools = self.TraitsImplementing<AmmoPool>();
+		}
 
 		public override Activity Tick(Actor self)
 		{
 			if (!target.IsValidFor(self))
 				return NextActivity;
 
-			var limitedAmmo = self.TraitOrDefault<LimitedAmmo>();
-			if (limitedAmmo != null && !limitedAmmo.HasAmmo())
+			// Move to the next activity only if all ammo pools are depleted and none reload automatically
+			if (ammoPools != null && ammoPools.All(x => !x.Info.SelfReloads && !x.HasAmmo()))
```

Comment:
I think I've said this in my previous round of reviewing, but the comment got lost.  At some point, this should really check whether there's any ammo left that is suitable for the current target. Leave a `// TODO`?